app/models/authentication.rb

```diff
@@ -81,6 +81,18 @@ def raise_event(status, _message = nil)
     MiqEvent.raise_evm_event_queue(ci, "#{prefix}_auth_#{status}")
   end
 
+  def assign_values(options)
```

Comment:
The naming of this feels weird to me, but I can't think of anything better offhand.  It feels like it shouldn't be necessary as `attributes=` feels like it does most of this (except the funny Array handling, but maybe the caller can just handle that?)